rename and improve generic make (#6047)
kyma-project/kyma: common/makefiles/generic-make-go.md

By example:
When CI run`make release` the following steps are executed:
- rule `release` depends on rules `resolve dep-status verify build-image push-image`
- rule`resolve` does not appear in the Makefile. but it's generated. 
Notice this line of code:
```makefile
MOUNT_TARGETS = build resolve ensure dep-status check-imports imports check-fmt fmt errcheck vet generate pull-licenses gqlgen
$(foreach t,$(MOUNT_TARGETS),$(eval $(call buildpack-mount,$(t))))
```
for resolve, and others targets defined in `MOUNT_TARGETS`, function `buildpack-mount` is callled, which dynamically defines new rule:
```makefile
resolve:
    @echo make resolve
    @docker run $(DOCKER_INTERACTIVE) \
        -v $(COMPONENT_DIR):$(WORKSPACE_COMPONENT_DIR):delegated \
        $(DOCKER_CREATE_OPTS) make resolve-local
```
  as you can see, rule `resolve-local` is executed inside container.
- if `resolve` pass, then the next rule will be executed.
- if exit code of any rules is different than 0, makefile will abort execution and fail.

Target types:
- `MOUNT_TARGET` - mount component directory as volume to docker container
- `COPY_TAGRT` - copy components files to docker container.

## How to adjust makefile

> ### How to disable current rule in local makefile

Reviewer: ```suggestion
### Disable the current rule in the local Makefile
```

> In new Makefile add the following line: `{rule}: ;`.

Reviewer: ```suggestion
To disable a rule in the new Makefile, follow it with the semicolon `;`. For example, write: `{RULE}: ;`.
```

> There will be warnings printed on console, but the rule will be disabled.

Reviewer: ```suggestion
This results in the rule being disabled and warnings printed on the console.
```

### How to add new local rule, which doesn't need `BUILDPACK`:
Define rule in local makefile.
Add this rule to one of the  global rule:
```makefile
verify:: own-rule
```

### How to add new rule in local makefile, which needs buildpack:
Define rule in local makefile and call function which will create the rule:
```makefile
my-rule-local: 
    do sth

$(eval $(call BUILDPACK_FUNCTION,my-rule)) # function which will create the new rule
```

Available BUILDPACK_FUNCTIONS:
- buildpack-mount
- buildpkac-cp-ro

> ### How to add new rule in generic makefile:

Reviewer: ```suggestion
### Add a new rule in the Generic Makefile
```
